NebulaServer/GameLogic/PlayerManager.cs

```diff
@@ -80,5 +127,37 @@ public void PlayerDisconnected(NebulaConnection conn)
                 connectedPlayers.Remove(conn);
             }
         }
+
+        public void PlayerSentSyncComplete(Player player)
+        {
+            syncingPlayers.Remove(player.connection);
+            connectedPlayers.Add(player.connection, player);
+
+            stateServer.Stop();
+            stateServer = null;
+
+            // Signal to all the other users that they can now unpause
+            SendPacketToOtherPlayers(new SyncComplete(), player);
+
+            // Send a confirmation to the new player contaning his player id.
+            player.SendPacket(new JoinSessionConfirmed(player.Id));
+        }
+
+        public void PlayerSentInitialState(Player player, InitialState packet)
+        {
+            if(!player.IsMasterClient)
+            {
+                // Someone is doing something nefarious here
+                return;
+            }
+
+            stateServer = new HttpStateServer($"http://{ Server.ServerIP }:17432", packet.URI);
```

Comment:
We could use the same port since the HTTP server is TCP and the game server UDP. That would have the benefit to only need to port forward 1 port instead of 2.